pallets/pallet-lottery/src/tests.rs

```diff
@@ -0,0 +1,271 @@
+		let alice_won: Balance = Percent::from_percent(20) * total;
+		let dave_tax: Balance = TAX_RATE * dave_won;
+		let bob_tax: Balance = TAX_RATE * bob_won;
+		let alice_tax: Balance = TAX_RATE * alice_won;
+
+		// Select the winner
+		Lottery::on_finalize(LOTTERY_PAYOUT_PERIOD);
+
+		// Verify data
+		assert_eq!(Bank::free_balance(&ALICE), 1_000 * DOLLAR + alice_won - alice_tax);
+		assert_eq!(Bank::free_balance(&BOB), 1_000 * DOLLAR + bob_won - bob_tax);
+		assert_eq!(Bank::free_balance(&CHARLIE), 1_000 * DOLLAR);
+		assert_eq!(Bank::free_balance(&DAVE), 1_000 * DOLLAR + dave_won - dave_tax);
+
+		assert_eq!(Bank::free_balance(&PRIZE_POOL_ACCOUNT), Default::default());
+		assert_eq!(Bank::free_balance(&TREASURY), TAX_RATE * total);
+
+		System::assert_has_event(RuntimeEvent::Lottery(Event::<Runtime>::LotteryWon {
+			user: DAVE,
+			won_fund: dave_won - dave_tax,
+			tax: dave_tax,
+		}));
+		System::assert_has_event(RuntimeEvent::Lottery(Event::<Runtime>::LotteryWon {
+			user: BOB,
+			won_fund: bob_won - bob_tax,
+			tax: bob_tax,
+		}));
+		System::assert_last_event(RuntimeEvent::Lottery(Event::<Runtime>::LotteryWon {
+			user: ALICE,
+			won_fund: alice_won - alice_tax,
+			tax: alice_tax,
+		}));
+	});
+}
+
+#[test]
+fn not_enough_players_works() {
+	let mock = MockGenesisConfig { balance: vec![], lotteries: vec![] };
+	mock.with_balances(vec![(ALICE, 1_000 * DOLLAR), (BOB, 1_000 * DOLLAR)])
+		.with_lotteries(vec![(ALICE, 100), (BOB, 100)])
+		.build()
+		.execute_with(|| {
+			RandomOutput::set(10u32);
+
+			// Set prize split
+			let split = vec![
+				Percent::from_percent(50),
+				Percent::from_percent(30),
+				Percent::from_percent(20),
+			];
+			assert_ok!(Lottery::set_prize_split(RuntimeOrigin::root(), split));
+
+			// Manager Eve initialize ticket price $1, so that the pool has $20.
+			// Storage Map iterate order is random but deterministic.
+			// In this case the iterate order is Bob Alice
+			let total: Balance = 200 * DOLLAR;
+			let bob_won: Balance = Percent::from_percent(50) * total;
+			let alice_won: Balance = Percent::from_percent(30) * total;
+			let bob_tax: Balance = TAX_RATE * bob_won;
+			let alice_tax: Balance = TAX_RATE * alice_won;
+
+			// Select the winner
+			Lottery::on_finalize(LOTTERY_PAYOUT_PERIOD);
+
+			// Verify data
+			assert_eq!(Bank::free_balance(&ALICE), 1_000 * DOLLAR + alice_won - alice_tax);
+			assert_eq!(Bank::free_balance(&BOB), 1_000 * DOLLAR + bob_won - bob_tax);
+
+			assert_eq!(Bank::free_balance(&PRIZE_POOL_ACCOUNT), total - alice_won - bob_won);
+			assert_eq!(Bank::free_balance(&TREASURY), alice_tax + bob_tax);
+
+			System::assert_has_event(RuntimeEvent::Lottery(Event::<Runtime>::LotteryWon {
+				user: BOB,
+				won_fund: bob_won - bob_tax,
+				tax: bob_tax,
+			}));
+			System::assert_last_event(RuntimeEvent::Lottery(Event::<Runtime>::LotteryWon {
+				user: ALICE,
+				won_fund: alice_won - alice_tax,
+				tax: alice_tax,
+			}));
+		});
+}
```

Comment:
Should add more tests for the "select_winner" function. Systematically test the correct winner can be selected depending on the random number.